xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/DefaultModelBridge.java

```diff
@@ -179,15 +179,26 @@ public boolean copy(DocumentReference source, DocumentReference destination)
 
     @Override
     public boolean delete(DocumentReference reference)
+    {
+        return delete(reference, true);
+    }
+
+    @Override
+    public boolean expurge(DocumentReference reference)
+    {
+        return delete(reference, false);
+    }
+
+    private boolean delete(DocumentReference reference, boolean totrash)
     {
         XWikiContext xcontext = this.xcontextProvider.get();
         try {
             XWikiDocument document = xcontext.getWiki().getDocument(reference, xcontext);
             if (document.getTranslation() == 1) {
-                xcontext.getWiki().deleteDocument(document, xcontext);
+                xcontext.getWiki().deleteDocument(document, totrash, xcontext);
                 this.logger.info("Document [{}] has been deleted.", reference);
```

Comment:
I would put the toTrash value in the logger info too.

Reply:
Note that we don't display "info" level messages in the logs except for very important things, mostly at startup. I know it's not in your commit but I don't see this info() logging in the sources on master.

Reply:
Maybe you missed it, it's there: https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/DefaultModelBridge.java#L188

Reply:
> Note that we don't display "info" level messages in the logs except for very important things

This one generally does not end up in the general log because it's used in jobs.

Reply:
DefaultModelBridge definitely doesn't start any job so it would appear in logs for any script calling the method. The logging should be here.

Reply:
I don't understand what you mean. This is an internal component used in refactoring jobs, not a script service.

Reply:
oh sorry.... I got tricked by the class name.... I thought this was `com.xpn.xwiki.internal.sheet.DefaultModelBridge` (`public class DefaultModelBridge implements ModelBridge`)... :)

Reply:
Yes we have quite a collection of "DefaultModelBridge" classes.

Reply:
Updated

Reply:
@manuelleduc  shouldn't the logging info also mention if the doc has been put in the recycle bin or not?

Reply:
I have update the logs with `(send to trash: [{}])`, displaying `totrash` value. See https://github.com/xwiki/xwiki-platform/pull/1369/files#diff-29fdf5e5e3805db76947ad7e90902ac7R199